src/main/java/edu/ucla/library/bucketeer/verticles/MainVerticle.java

```diff
@@ -47,73 +50,91 @@
 
     private static final int DEFAULT_S3_UPLOADER_INSTANCES = 1;
 
-    private static final int DEFAULT_S3_UPLOADER_THREADS = 20;
+    private static final int DEFAULT_S3_UPLOADER_THREADS = getAvailableProcessors();
+
+    private static final String EMPTY = "";
+
+    /**
+     * Gets the available logical processors on the system minus one.
+     *
+     * @return
+     */
+    private static int getAvailableProcessors() {
+        final SystemInfo systemInfo = new SystemInfo();
+        final HardwareAbstractionLayer hardwareAbstractionLayer = systemInfo.getHardware();
+        final CentralProcessor centralProcessor = hardwareAbstractionLayer.getProcessor();
+
+        return centralProcessor.getLogicalProcessorCount() - 1;
```

Comment:
Seems like we're going to -1 core route where we'd want to leave a core free for processing/adding items into queues?

Reply:
wondering if there's an edge case here where we might get a 0 - 1 for core count.

Reply:
As a default, I went core -1. I think we need to try more threads as an option though. I'm not sure whether the cost of thread management will outweigh the additional benefit of thread swapping. I can't imagine a machine that has zero logical cores. I could imagine a single core machine, but not sure what the chance of that would be.